integration_tests/src/main/python/collect_limit_test.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from data_gen import *
+from asserts import assert_gpu_and_cpu_are_equal_collect
+
+allow_collect_conf={"spark.rapids.sql.exec.CollectLimitExec": "true"}
+
+@pytest.mark.parametrize('data_gen', all_gen)
+def test_collect_limit(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(lambda spark: unary_op_df(spark, data_gen).limit(10),
```

Comment:
Do we have issues with this test when there are multiple shuffle partitions?  It doesn't seem deterministic as written.

Reply:
I ran the test a couple of times locally without a problem. Let me add a `repartition` before the limit.

Reply:
So, the moment I add a `repartition` it starts to fail like you suspected. If I add `orderBy` or `sort`, `CollectLimitExec` is not executed at all. So, I have now made the limit = length of df, let me know if this is good. 

Any other suggestions to make the test better will be appreciated

Reply:
This type of a test is going to be non-deterministic.  If we fixed the GPU version to ramp up like the CPU version does, then it would be much more likely to be deterministic, but until then I am fine with us just not having an automated test. We didn't have one before. The code is disabled by default. Lets just manually test it.

Reply:
what do you think about adding a scala test instead?